# 1. Load Data and Import Library

In [1]:
from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [2]:
import pandas as pd
import numpy as np
from google.colab import files

In [3]:
episode = pd.read_csv('/content/drive/MyDrive/Bootcamp/PROJECT/F.R.I.E.N.D.S/friends_episodes.csv')
imdb = pd.read_csv('/content/drive/MyDrive/Bootcamp/PROJECT/F.R.I.E.N.D.S/friends_imdb.csv')

In [4]:
episode.head(5)

season  episode_num_in_season  episode_num_overall  \
0       1                      1                    1   
1       1                      2                    2   
2       1                      3                    3   
3       1                      4                    4   
4       1                      5                    5   

                                            title    directed_by  \
0            The One Where Monica Gets a Roommate  James Burrows   
1            The One with the Sonogram at the End  James Burrows   
2                          The One with the Thumb  James Burrows   
3              The One with George Stephanopoulos  James Burrows   
4  The One with the East German Laundry Detergent  Pamela Fryman   

                       written_by original_air_date prod_code  us_viewers  
0    David Crane & Marta Kauffman        1994-09-22    456650  21500000.0  
1    David Crane & Marta Kauffman        1994-09-29    456652  20200000.0  
2  Jeffrey Astrof & Mike Sikowitz        1994-10-06    456651  19500000.0  
3                     Alexa Junge        1994-10-13    456654  19700000.0  
4  Jeff Greenstein & Jeff Strauss        1994-10-20    456653  18600000.0

In [5]:
imdb.head(5)

season  episode_num                                           title  \
0       1            1            The One Where Monica Gets a Roommate   
1       1            2            The One with the Sonogram at the End   
2       1            3                          The One with the Thumb   
3       1            4              The One with George Stephanopoulos   
4       1            5  The One with the East German Laundry Detergent   

  original_air_date  imdb_rating  total_votes  \
0      22 Sep. 1994          8.3         8378   
1      29 Sep. 1994          8.0         6441   
2       6 Oct. 1994          8.1         6060   
3      13 Oct. 1994          8.1         5892   
4      20 Oct. 1994          8.4         5872   

                                                desc  
0  Monica and the gang introduce Rachel to the "r...  
1  Ross finds out his ex-wife is pregnant. Rachel...  
2  Monica becomes irritated when everyone likes h...  
3  Joey and Chandler take Ross to a hockey game t...  
4  Eager to spend time with Rachel, Ross pretends...

In [6]:
episode.tail(5)

season  episode_num_in_season  episode_num_overall  \
231      10                     14                  232   
232      10                     15                  233   
233      10                     16                  234   
234      10                     17                  235   
235      10                     18                  236   

                                      title      directed_by  \
231          The One with Princess Consuela   Gary Halvorson   
232              The One Where Estelle Dies   Gary Halvorson   
233  The One with Rachel's Going Away Party   Gary Halvorson   
234                            The Last One  Kevin S. Bright   
235                            The Last One  Kevin S. Bright   

                                            written_by original_air_date  \
231  Story by: Robert CarlockTeleplay by: Tracy Reilly        2004-02-26   
232  Story by: Mark KunerthTeleplay by: David Crane...        2004-04-22   
233                           Andrew Reich & Ted Cohen        2004-04-29   
234                       Marta Kauffman & David Crane        2004-05-06   
235                       Marta Kauffman & David Crane        2004-05-06   

    prod_code  us_viewers  
231    176263  22820000.0  
232    176264  22640000.0  
233    176265  24510000.0  
234    176266  52460000.0  
235    176267  52460000.0

In [7]:
imdb.tail(5)

season  episode_num                                   title  \
230      10           13        The One Where Joey Speaks French   
231      10           14          The One with Princess Consuela   
232      10           15              The One Where Estelle Dies   
233      10           16  The One with Rachel's Going Away Party   
234      10           17                            The Last One   

    original_air_date  imdb_rating  total_votes  \
230      19 Feb. 2004          8.5         4054   
231      26 Feb. 2004          8.6         3953   
232      22 Apr. 2004          8.5         3700   
233      29 Apr. 2004          8.9         4290   
234        6 May 2004          9.7        14033   

                                                  desc  
230  Phoebe tries to teach Joey to speak French for...  
231  When Phoebe goes to get her name changed she r...  
232  Ross tries to get Rachel to go back to Ralph L...  
233  The gang throws Rachel a goodbye party, during...  
234  Phoebe races Ross to the airport in a bid to s...

Since episode 235 and 236 is the same episode, it will be more simple if we drop one of them.

# 2. Data Cleansing & Data Manipulation

In [8]:
episode = episode.drop(235)

In [9]:
episode.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 235 entries, 0 to 234
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   season                 235 non-null    int64  
 1   episode_num_in_season  235 non-null    int64  
 2   episode_num_overall    235 non-null    int64  
 3   title                  235 non-null    object 
 4   directed_by            235 non-null    object 
 5   written_by             235 non-null    object 
 6   original_air_date      235 non-null    object 
 7   prod_code              235 non-null    object 
 8   us_viewers             235 non-null    float64
dtypes: float64(1), int64(3), object(5)
memory usage: 18.4+ KB


In [10]:
imdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 235 entries, 0 to 234
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   season             235 non-null    int64  
 1   episode_num        235 non-null    int64  
 2   title              235 non-null    object 
 3   original_air_date  235 non-null    object 
 4   imdb_rating        235 non-null    float64
 5   total_votes        235 non-null    int64  
 6   desc               235 non-null    object 
dtypes: float64(1), int64(3), object(3)
memory usage: 13.0+ KB


In [11]:
episode = episode.convert_dtypes()
episode.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 235 entries, 0 to 234
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   season                 235 non-null    Int64  
 1   episode_num_in_season  235 non-null    Int64  
 2   episode_num_overall    235 non-null    Int64  
 3   title                  235 non-null    string 
 4   directed_by            235 non-null    string 
 5   written_by             235 non-null    string 
 6   original_air_date      235 non-null    string 
 7   prod_code              235 non-null    string 
 8   us_viewers             235 non-null    Float64
dtypes: Float64(1), Int64(3), string(5)
memory usage: 19.3 KB


In [12]:
episode['original_air_date'] = pd.to_datetime(episode['original_air_date'])
episode.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 235 entries, 0 to 234
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   season                 235 non-null    Int64         
 1   episode_num_in_season  235 non-null    Int64         
 2   episode_num_overall    235 non-null    Int64         
 3   title                  235 non-null    string        
 4   directed_by            235 non-null    string        
 5   written_by             235 non-null    string        
 6   original_air_date      235 non-null    datetime64[ns]
 7   prod_code              235 non-null    string        
 8   us_viewers             235 non-null    Float64       
dtypes: Float64(1), Int64(3), datetime64[ns](1), string(4)
memory usage: 19.3 KB


In [13]:
imdb = imdb.convert_dtypes()
imdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 235 entries, 0 to 234
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   season             235 non-null    Int64  
 1   episode_num        235 non-null    Int64  
 2   title              235 non-null    string 
 3   original_air_date  235 non-null    string 
 4   imdb_rating        235 non-null    Float64
 5   total_votes        235 non-null    Int64  
 6   desc               235 non-null    string 
dtypes: Float64(1), Int64(3), string(3)
memory usage: 13.9 KB


In [14]:
imdb['original_air_date'] = pd.to_datetime(imdb['original_air_date'])
imdb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 235 entries, 0 to 234
Data columns (total 7 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   season             235 non-null    Int64         
 1   episode_num        235 non-null    Int64         
 2   title              235 non-null    string        
 3   original_air_date  235 non-null    datetime64[ns]
 4   imdb_rating        235 non-null    Float64       
 5   total_votes        235 non-null    Int64         
 6   desc               235 non-null    string        
dtypes: Float64(1), Int64(3), datetime64[ns](1), string(2)
memory usage: 13.9 KB


There are same 4 columns between episode and imdb dataframe (`season`, `episode_num`, `title`, and `original_air_date`). It will be more simple if we drop those columns.

In [15]:
# Remove 4 columns
imdb = imdb.drop(columns=['season','episode_num','title','original_air_date'])
imdb.head()

imdb_rating  total_votes                                               desc
0          8.3         8378  Monica and the gang introduce Rachel to the "r...
1          8.0         6441  Ross finds out his ex-wife is pregnant. Rachel...
2          8.1         6060  Monica becomes irritated when everyone likes h...
3          8.1         5892  Joey and Chandler take Ross to a hockey game t...
4          8.4         5872  Eager to spend time with Rachel, Ross pretends...

In [16]:
# Combine episode and imdb dataframes become one dataframe
friends = pd.concat([episode, imdb], axis=1)
friends.head()

season  episode_num_in_season  episode_num_overall  \
0       1                      1                    1   
1       1                      2                    2   
2       1                      3                    3   
3       1                      4                    4   
4       1                      5                    5   

                                            title    directed_by  \
0            The One Where Monica Gets a Roommate  James Burrows   
1            The One with the Sonogram at the End  James Burrows   
2                          The One with the Thumb  James Burrows   
3              The One with George Stephanopoulos  James Burrows   
4  The One with the East German Laundry Detergent  Pamela Fryman   

                       written_by original_air_date prod_code  us_viewers  \
0    David Crane & Marta Kauffman        1994-09-22    456650  21500000.0   
1    David Crane & Marta Kauffman        1994-09-29    456652  20200000.0   
2  Jeffrey Astrof & Mike Sikowitz        1994-10-06    456651  19500000.0   
3                     Alexa Junge        1994-10-13    456654  19700000.0   
4  Jeff Greenstein & Jeff Strauss        1994-10-20    456653  18600000.0   

   imdb_rating  total_votes                                               desc  
0          8.3         8378  Monica and the gang introduce Rachel to the "r...  
1          8.0         6441  Ross finds out his ex-wife is pregnant. Rachel...  
2          8.1         6060  Monica becomes irritated when everyone likes h...  
3          8.1         5892  Joey and Chandler take Ross to a hockey game t...  
4          8.4         5872  Eager to spend time with Rachel, Ross pretends...

In [17]:
friends.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 235 entries, 0 to 234
Data columns (total 12 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   season                 235 non-null    Int64         
 1   episode_num_in_season  235 non-null    Int64         
 2   episode_num_overall    235 non-null    Int64         
 3   title                  235 non-null    string        
 4   directed_by            235 non-null    string        
 5   written_by             235 non-null    string        
 6   original_air_date      235 non-null    datetime64[ns]
 7   prod_code              235 non-null    string        
 8   us_viewers             235 non-null    Float64       
 9   imdb_rating            235 non-null    Float64       
 10  total_votes            235 non-null    Int64         
 11  desc                   235 non-null    string        
dtypes: Float64(2), Int64(4), datetime64[ns](1), string(5)
memory usa

# 3. Export file into spreadsheet

In [18]:
friends.to_excel('friends.xls') 
files.download('friends.xls')

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: FutureWarning: As the xlwt package is no longer maintained, the xlwt engine will be removed in a future version of pandas. This is the only engine in pandas that supports writing in the xls format. Install openpyxl and write to an xlsx file instead. You can set the option io.excel.xls.writer to 'xlwt' to silence this warning. While this option is deprecated and will also raise a warning, it can be globally set and the warning suppressed.
  """Entry point for launching an IPython kernel.


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>